src/Microsoft.NetCore.Analyzers/Core/Security/UseDefaultDllImportSearchPathsAttribute.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseDefaultDllImportSearchPathsAttribute : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5392";
+        private static readonly LocalizableString s_Title = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.UseDefaultDllImportSearchPathsAttribute),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.UseDefaultDllImportSearchPathsAttributeMessage),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.UseDefaultDllImportSearchPathsAttributeDescription),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+
+        private const int UnsafeBits = 2 | 256;
```

Comment:
>2 | 256; [](start = 39, length = 8)

Add a comment for what these values are.

Given that we're treating AssemblyDirectory as potentially dangerous, should do the same for ApplicationDirectory.

Reply:
When using ApplicationDirectory, it won't search by the default order after it didn't find the target DLL in the application directory. When using AssemblyDirectory, it does, that's why we treat it as potentially dangerous.

Reply:
I think the attack scenario of executing an application with elevated privileges from a user's "Downloads" directory, which contains a malicious DLL, is something we should worry about.  In that case, I think ApplicationDirectory is also a problem, because it's going to include the "Downloads" directory.